### PR TITLE
Fix PSES Commands module

### PR DESCRIPTION
**Pull request by rjmholt**
Fixes https://github.com/PowerShell/vscode-powershell/issues/2495

### CODE REVIEW

File: module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1 [hunk 1]
```diff
@@ -36,18 +36,17 @@ function Register-EditorCommand {
     {
         $commandArgs = @($Name, $DisplayName, $SuppressOutput.IsPresent)
 
-        if ($ScriptBlock -ne $null)
+        $editorCommand = if ($ScriptBlock -ne $null)
         {
-            Write-Verbose "Registering command '$Name' which executes a ScriptBlock"
-            $commandArgs += $ScriptBlock
+            $null = Write-Verbose "Registering command '$Name' which executes a ScriptBlock"
```
Comment (TylerLeonhardt) [inline module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1]: why the `$null =`?

File: module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1 [hunk 2]
```diff
@@ -36,18 +36,17 @@ function Register-EditorCommand {
...
+            [Microsoft.PowerShell.EditorServices.Extensions.EditorCommand, Microsoft.PowerShell.EditorServices]::new($Name, $DisplayName, $SuppressOutput, $ScriptBlock)
         }
         else
         {
-            Write-Verbose "Registering command '$Name' which executes a function"
-            $commandArgs += $Function
+            $null = Write-Verbose "Registering command '$Name' which executes a function"
```
Comment (TylerLeonhardt) [inline module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1]: why the `$null =`?

Comment (rjmholt) [reply module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1]: Just as a guarantee that `Write-Verbose` won't emit any arguments. But if we already feel like that's guaranteed, I can take it out

Comment (SeeminglyScience) [reply module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1]: Yeah it doesn't call `WriteObject`

File: module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1 (hunk 1 above)
Comment (rjmholt) [inline module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1]:
```suggestion
            Write-Verbose "Registering command '$Name' which executes a ScriptBlock"
```

File: module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1 (hunk 2 above)
Comment (rjmholt) [inline module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1]:
```suggestion
            Write-Verbose "Registering command '$Name' which executes a function"
```

File: module/PowerShellEditorServices/Commands/Private/BuiltInCommands.ps1 [hunk 3]
```diff
@@ -3,7 +3,7 @@ Register-EditorCommand `
     -DisplayName 'Open Editor Profile' `
     -SuppressOutput `
     -ScriptBlock {
-        param([Microsoft.PowerShell.EditorServices.Services.PowerShellContext.EditorContext]$context)
+        param([Microsoft.PowerShell.EditorServices.Extensions.EditorContext, Microsoft.PowerShell.EditorServices]$context)
```
Comment (SeeminglyScience) [inline module/PowerShellEditorServices/Commands/Private/BuiltInCommands.ps1]: Does this syntax work with 5.1? I vaguely remember you running into issues where you needed the full qualification in 5.1.  Or was it that the full qualification didn't work but this did?

Comment (rjmholt) [reply module/PowerShellEditorServices/Commands/Private/BuiltInCommands.ps1]: Ah good thinking! Now that you mention it, I remember a similar issue (with PSRL type invocation).

But I just tested it and this works in 5.1:

```
PS C:\Users\Robert Holt\Documents\Dev\sandbox> [microsoft.powershell.editorservices.extensions.editorcommand, microsoft.powershell.editorservices]


IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     False    EditorCommand                            System.Object
```

**Pull request merged**